.github/actions/markPullRequestsAsDeployed/markPullRequestsAsDeployed.js

```diff
@@ -30,18 +33,92 @@ function getDeployTableMessage(platformResult) {
     }
 }
 
+/**
+ * Get PR's neccessary information
+ *
+ * @param {Number} [pr] PR number
+ * @returns {Promise<any>}
+ */
+function getPR(pr) {
+    return GithubUtils.octokit.pulls.get({
+        owner: GithubUtils.GITHUB_OWNER,
+        repo: GithubUtils.GITHUB_REPOSITORY,
+        pull_number: pr,
+    }).then(({data}) => ({
+        mergedAt: data.merged_at,
+        hasCPStagingLabel: _.contains(_.pluck(data.labels, 'name'), 'CP Staging'),
+    }));
+}
+
+/**
+ * Get the [added, removed] pairs for the `🔐 LockCashDeploys 🔐` label on StagingDeployCash
+ *
+ * @return {Promise<Array<[string, string]>>}
+ */
+function getLockCashDeploysTimeline() {
+    return GithubUtils.octokit.paginate(GithubUtils.octokit.issues.listEvents, {
+        owner: GithubUtils.GITHUB_OWNER,
+        repo: GithubUtils.GITHUB_REPOSITORY,
+        issue_number: stagingDeployIssueNumber,
+        per_page: 100,
+    }).then((events) => {
+        let pair = [];
+        const startEndPairs = _.compact(_.map(events, ({event, created_at, label}, index) => {
+            if (event === 'labeled' && label.name === '🔐 LockCashDeploys 🔐') {
+                if (pair.length) {
+                    // flush the pair
+                    pair = [];
+                }
+                pair.push(created_at);
+            } else if (event === 'unlabeled' && label.name === '🔐 LockCashDeploys 🔐') {
+                pair.push(created_at);
+            }
+            if (index === events.length - 1 && pair.length === 1) {
+                pair.push(moment().toISOString());
+                return pair;
+            }
+            return pair.length > 1 ? pair : undefined;
+        }));
+        return startEndPairs;
+    });
+}
+
 const androidResult = getDeployTableMessage(core.getInput('ANDROID', {required: true}));
 const desktopResult = getDeployTableMessage(core.getInput('DESKTOP', {required: true}));
 const iOSResult = getDeployTableMessage(core.getInput('IOS', {required: true}));
 const webResult = getDeployTableMessage(core.getInput('WEB', {required: true}));
 
 const workflowURL = `${process.env.GITHUB_SERVER_URL}/${process.env.GITHUB_REPOSITORY}`
     + `/actions/runs/${process.env.GITHUB_RUN_ID}`;
-const deployVerb = isCP ? 'Cherry-picked' : 'Deployed';
 
-let message = `🚀 [${deployVerb}](${workflowURL}) to ${isProd ? 'production' : 'staging'} in version: ${version}🚀`;
-message += `\n\n platform | result \n ---|--- \n🤖 android 🤖|${androidResult} \n🖥 desktop 🖥|${desktopResult}`;
-message += `\n🍎 iOS 🍎|${iOSResult} \n🕸 web 🕸|${webResult}`;
+/**
+ * Get Deploy Verb for the PR
+ *
+ * @param {Number} pr
+ * @return {Promise<'Cherry-picked' | 'Deployed'>}
+ */
+function getPRDeployVerb(pr) {
+    return getPR(pr).then(({mergedAt, hasCPStagingLabel}) => {
+        if (!hasCPStagingLabel) {
```

Comment:
So is it true that if there is no CP staging label, then normal Deploy?

Reply:
Yep, that's true!